django/db/models/sql/query.py

```diff
@@ -1639,6 +1641,8 @@ def split_exclude(self, filter_expr, can_reuse, names_with_path):
         saner null handling, and is easier for the backend's optimizer to
         handle.
         """
+        if isinstance(filter_expr[1], F):
```

Comment:
Would this improve readability rather than using "magic indexing"?
`filter_lhs, filter_rhs = filter_expr`